drivers/mtd/mtd.c

```diff
@@ -50,6 +50,21 @@ int mtd_read(mtd_dev_t *mtd, void *dest, uint32_t addr, uint32_t count)
     }
 }
 
+int mtd_read_page(mtd_dev_t *mtd, void *dest, uint32_t page, uint16_t offset, uint32_t count)
+{
+    if (!mtd || !mtd->driver) {
+        return -ENODEV;
+    }
+
```

Comment:
It would be super nice of have a range check here, so that the MTD backends could simply assume never being called with out of range read, write or erase requests. However, it seems this would require a bunch of multiplication for each call to `mtd_read_page()` to be able to do so :-/

Reply:
Yea I also noticed that with the current MTD API, the burden is on the *file system* adaption code to ensure writes to not exceed a page and implement the chunking.
This is of course a silly duplication and the point of the intermediate layer should be that the consumers don't have to care about the implementation details.

That's why I added WIP - I'll change the low-level `write_page` function to operate like the internal [`_write_page`](https://github.com/RIOT-OS/RIOT/blob/master/drivers/at25xxx/at25xxx.c#L84) in `at25xxx.c`. Write as many bytes as fit into the current page, don't try anything clever and return the number of written bytes.

`mtd_write_page()` will then use that to split up however many bytes the user wants to write (like [`at25xxx_write()`](https://github.com/RIOT-OS/RIOT/blob/master/drivers/at25xxx/at25xxx.c#L125)

That way we can pull that logic out of the drivers and into a common place - and users of the MTD API don't have to care about the properties of the underlying storage media.